clusterloader2/state/namespaces.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"fmt"
+	"sync"
+
+	"k8s.io/perf-tests/clusterloader2/api"
+)
+
+// InstancesState represents state of object replicas.
+type InstancesState struct {
+	Number int32
+	Object api.Object
+}
+
+type instancesIdentifier struct {
+	objectKind string
```

Comment:
That is still not enough.
In theory, nothing prevents you from having the same Kind in different api groups.

Reply:
OK. Can you have same kind and api group, but different api version?

Reply:
No - objects should be convertible between versions.

Reply:
ok. So I'm adding only api groups.